### PR TITLE
Deny panic, unwrap and expect in the spec module

### PR DESCRIPTION
**Pull request by o0Ignition0o**
### Deny unwrap and expect in the spec module ([Issue #1844](https://github.com/apollographql/router/pull/1844))

As we are progressing towards 1.0, we are progressively banning `unwrap()` and `expect()` from the critical parts of the codebase.

This codepath is exercised after Parsing has happened, so invariants expressed by `expect`s would always have been enforced or checked before. However we decided to tighten the router even further, by raising errors instead, which will provide users with even more stability guarantees.

The spec module now has gates that prevent its content from using code that explicitly panics.

```rust
#![deny(clippy::unwrap_used)]
#![deny(clippy::expect_used)]
#![deny(clippy::panic)]
```

### CODE REVIEW

**Review by garypen — APPROVED** [review]
It would probably be nice to consolidate all these new strings at some point. That can wait. Great improvement!

File: apollo-router/src/spec/schema.rs [hunk 1]
```diff
@@ -308,23 +317,31 @@ impl Schema {
...
+                                match <$ty>::try_from(definition) {
+                                    Ok(instance) => Some(Ok((instance.name.clone(), instance))),
+                                    Err(e) => Some(Err(e)),
+                                }
                             } else {
                                 None
                             }
                         })
-                        .collect::<HashMap<String, $ty>>();
+                        // todo: impl from
```
Comment (bnjjj) [inline apollo-router/src/spec/schema.rs:328]: Is it still relevant ? Or should we delete this comment ?

File: apollo-router/src/spec/schema.rs [hunk 2]
```diff
@@ -308,23 +317,31 @@ impl Schema {
...
                         .definitions()
                         .filter_map(|definition| {
                             if let $ast_extension_ty(extension) = definition {
-                                Some(<$ty>::from(extension))
+                                Some(<$ty>::try_from(extension))
                             } else {
                                 None
                             }
                         })
+                        // todo: impl from
```
Comment (bnjjj) [inline apollo-router/src/spec/schema.rs:341]: Should we delete it ?

**Review by bnjjj — APPROVED** [review]

File: apollo-router/src/spec/schema.rs (hunk 2 above)
Comment (o0Ignition0o) [reply apollo-router/src/spec/schema.rs:341]: Yeah, we ll deal with it once we refactor this bit of code away!

**Pull request merged**